overlord/snapstate/snapstate.go

```diff
@@ -2312,6 +2312,23 @@ func GadgetInfo(st *state.State, deviceCtx DeviceContext) (*snap.Info, error) {
 	return infoForDeviceSnap(st, deviceCtx, "gadget", (*asserts.Model).Gadget)
 }
 
+// KernelInfo finds the kernel snap's info for the given device context.
+func KernelInfo(st *state.State, deviceCtx DeviceContext) (*snap.Info, error) {
+	return infoForDeviceSnap(st, deviceCtx, "kernel", (*asserts.Model).Kernel)
+}
+
+// BootBaseInfo finds the boot base snap's info for the given device context.
+func BootBaseInfo(st *state.State, deviceCtx DeviceContext) (*snap.Info, error) {
+	baseName := func(mod *asserts.Model) string {
+		base := mod.Base()
+		if base == "" {
```

Comment:
There doesn't seem to be any unit tests covering, but since it's UC20 context that probably doesn't matter much.

Reply:
Ignore that. The `go-carpet` output tells me it's covered. Looked at the tests again, and there's a test with classic model too.